### PR TITLE
TCES-53 TIM-Timeline Window

### PR DESCRIPTION
**Pull request by selintasman**
Implemented the timeline windows for employer, client, and job lead.
Employer:
Screenshot 2024-01-30 at 9 16 01 PM
Client:
Screenshot 2024-01-30 at 9 15 47 PM
Job Lead:
Screenshot 2024-01-30 at 9 16 12 PM

Screenshot 2024-01-30 at 9 23 08 PM
Screenshot 2024-01-30 at 9 22 57 PM

### CODE REVIEW

File: client/src/components/timeline-window-component/clientWindow.jsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      fullWidth
+      size="small"
+      InputProps={{
+        startAdornment: (
+          <InputAdornment position="start">
+            <SearchIcon style={{ fontSize: '2rem' }}/>
+          </InputAdornment>
+        ),
+      }}
+    />
```
Comment (DanielDervishi) [inline client/src/components/timeline-window-component/clientWindow.jsx]: The search component can be removed from each of the timeline windows (It's been listed as a nice to have for now)

File: client/src/components/timeline-window-component/clientWindow.jsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+           
+            <TabContext value={value}>
+              <TabList onChange={handleChange} sx={{
+        '& .MuiTabs-flexContainer': { display: 'flex', width: "100%", gap: -2 }, 
+        '& .MuiButtonBase-root': {padding: 0}
+      }}>
+                <Tab label="ALL" value="1" />
+                <Tab label="CONTACTED" value="2" />
+                <Tab label="UPDATES" value="3" />
+                <Tab label="NOTES" value="4" />
```
Comment (DanielDervishi) [inline client/src/components/timeline-window-component/clientWindow.jsx]: The spacing between different tabs here looks a bit uneven
image

File: client/package.json [hunk 3]
```diff
@@ -6,8 +6,8 @@
     "@emotion/react": "^11.11.1",
```
Comment (DanielDervishi) [inline client/package.json:6]: Noticed a small bug when clicking the three dots at the top of the timeline window

https://github.com/uoftblueprint/tces/assets/58835213/22d595bb-8e64-4e24-8b40-4d97e062683d

**Review by DanielDervishi — COMMENTED** [review]
Great work Selin! The pr should be ready to be merged after these comments are addressed

File: client/src/components/timeline-window-component/clientWindow.jsx
Comment (3milyfz) [inline client/src/components/timeline-window-component/clientWindow.jsx:1]: image

I think you may have forgotten to run `docker compose run react-app npm run format`

**Review by 3milyfz — CHANGES_REQUESTED** [review]
Great job Selin! Just some minor formatting issues in terminal.

**Review by 3milyfz — DISMISSED** [review]

**Review by DanielDervishi — DISMISSED** [review]
Thanks for getting this done Selin!

File: client/src/App.jsx [hunk 4]
```diff
@@ -35,7 +35,6 @@ import EditJobLead from "./pages/edit-job-lead";
 // helper functions
```
Comment (DanielDervishi) [inline client/src/App.jsx]: Be sure to match the state of app.jsx on your branch with the state of it on main

**Review by DanielDervishi — APPROVED** [review]

**Review by 3milyfz — APPROVED** [review]

**Pull request merged**